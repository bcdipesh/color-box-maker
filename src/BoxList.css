.BoxList {
  margin: 10% auto;
}
