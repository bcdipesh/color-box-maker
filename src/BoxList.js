import { useState } from "react";
import "./BoxList.css";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]);

  const addBoxes = (box) => {
    setBoxes([...boxes, box]);
  };

  const removeBox = (boxId) => {
    setBoxes(boxes.filter((box, idx) => idx !== boxId));
  };

  return (
    <div className="BoxList">
      <NewBoxForm addBoxes={addBoxes} />
      {boxes.map(({ backgroundColor, width, height }, idx) => (
        <Box
          key={idx}
          boxId={idx}
          backgroundColor={backgroundColor}
          width={width}
          height={height}
          removeBox={removeBox}
        />
      ))}
    </div>
  );
};

export default BoxList;
