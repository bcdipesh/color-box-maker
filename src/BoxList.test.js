import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

const addBox = (boxList, height = "100px", width = "100px", color = "red") => {
  const heightInput = boxList.getByLabelText("Box Height");
  const widthInput = boxList.getByLabelText("Box Width");
  const backgroundColorInput = boxList.getByLabelText("Box Background Color");
  fireEvent.change(backgroundColorInput, { target: { value: color } });
  fireEvent.change(widthInput, { target: { value: width } });
  fireEvent.change(heightInput, { target: { value: height } });
  const addBoxBtn = boxList.getByText("Add Box");
  fireEvent.click(addBoxBtn);
};

it("renders without crashing", () => {
  render(<BoxList />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new box", () => {
  const boxList = render(<BoxList />);

  expect(boxList.queryByText("X")).not.toBeInTheDocument();

  addBox(boxList);

  const removeButton = boxList.getByText("X");
  expect(removeButton).toBeInTheDocument();
  expect(removeButton.parentElement).toHaveStyle(`
  width: 100px;
  height: 100px;
  background-color: red;`);
  expect(boxList.getAllByDisplayValue("")).toHaveLength(3);
});
