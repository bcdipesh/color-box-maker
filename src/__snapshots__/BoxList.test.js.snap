// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <div
      class="NewBoxForm"
    >
      <form>
        <div
          class="NewBoxForm-form-group"
        >
          <label
            for="width"
          >
            Box Width
          </label>
          <input
            id="width"
            name="width"
            placeholder="Box Width. Eg; 1px, 1rem, 1%"
            value=""
          />
        </div>
        <div
          class="NewBoxForm-form-group"
        >
          <label
            for="height"
          >
            Box Height
          </label>
          <input
            id="height"
            name="height"
            placeholder="Box Height: Eg; 1px, 1rem, 1%"
            value=""
          />
        </div>
        <div
          class="NewBoxForm-form-group"
        >
          <label
            for="backgroundColor"
          >
            Box Background Color
          </label>
          <input
            id="backgroundColor"
            name="backgroundColor"
            placeholder="Box Background Color: Eg; 1px, 1rem, 1%"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          Add Box
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
